tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -83,13 +83,21 @@ def test__delete_private_collection_by_collection_version_id(self):
                 private_collection_version_id = self.generate_unpublished_collection().version_id
                 self._test(private_collection_version_id, auth, expected_response)
 
-    def test__delete_tombstone_collection(self):
+    @patch("backend.layers.thirdparty.s3_provider.S3Provider.delete_files")  # required for tombstoning
+    def test__delete_tombstone_collection(self, mock_delete: Mock):
         tests = [("not_owner", 403), ("noauth", 401), ("owner", 403), ("super", 403)]
         for auth, expected_response in tests:
             with self.subTest(auth):
-                collection_id = self.generate_published_collection().collection_id
-                self.business_logic.tombstone_collection(collection_id)
-                self._test(collection_id, auth, expected_response)
+                collection = self.generate_published_collection()
+                self.business_logic.tombstone_collection(collection.collection_id)
+                if self.run_as_integration:
+                    dataset_version_ids = [d_v.version_id.id for d_v in collection.datasets]
+                    delete_args = set(mock_delete.call_args.args[1])
+                    expected_delete_args = set()
+                    for file_type in ("h5ad", "rds"):
+                        expected_delete_args.update([f"{d_v_id}.{file_type}" for d_v_id in dataset_version_ids])
```

Comment:
I will be doing the dataset file structure migration before this hits prod followed by mirrors from prod to dev and staging, so this will work (i.e. every dataset file will, at that point, live at a `<dataset_version_id>.<file_type>` key)